debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -179,31 +181,12 @@ private void readChangeStream(MongoClient client, MongoPreferredNode mongo, Repl
         final ServerAddress nodeAddress = MongoUtil.getPreferredAddress(client, mongo.getPreference());
         LOGGER.info("Reading change stream for '{}'/{} from {} starting at {}", replicaSet, mongo.getPreference().getName(), nodeAddress, oplogStart);
 
-        final List<Bson> pipeline = new ChangeStreamPipelineFactory(rsOffsetContext, taskContext.getConnectorConfig(), taskContext.filters().getConfig()).create();
-        final ChangeStreamIterable<BsonDocument> rsChangeStream = client.watch(pipeline, BsonDocument.class);
-        if (taskContext.getCaptureMode().isFullUpdate()) {
-            rsChangeStream.fullDocument(FullDocument.UPDATE_LOOKUP);
-        }
-        if (taskContext.getCaptureMode().isIncludePreImage()) {
-            rsChangeStream.fullDocumentBeforeChange(FullDocumentBeforeChange.WHEN_AVAILABLE);
-        }
-        if (rsOffsetContext.lastResumeToken() != null) {
-            LOGGER.info("Resuming streaming from token '{}'", rsOffsetContext.lastResumeToken());
-
-            final BsonDocument doc = new BsonDocument();
-            doc.put("_data", new BsonString(rsOffsetContext.lastResumeToken()));
-            rsChangeStream.resumeAfter(doc);
-        }
-        else if (oplogStart.getTime() > 0) {
-            LOGGER.info("Resume token not available, starting streaming from time '{}'", oplogStart);
-            rsChangeStream.startAtOperationTime(oplogStart);
-        }
-
-        if (connectorConfig.getCursorMaxAwaitTime() > 0) {
-            rsChangeStream.maxAwaitTime(connectorConfig.getCursorMaxAwaitTime(), TimeUnit.MILLISECONDS);
-        }
+        var rsChangeStream = createChangeStream(client, rsOffsetContext);
+        var cursor = rsChangeStream.cursor();
 
-        try (MongoChangeStreamCursor<ChangeStreamDocument<BsonDocument>> cursor = rsChangeStream.cursor()) {
+        // Once every 10 seconds
+        var preferenceRateLimiter = new RateLimiter(1, 0.1F);
```

Comment:
Should this be configurable?

Reply:
Though I do like this as it is the standard leaky bucket algorithm, there already is `io.debezium.util.DelayStrategy`, look whether that can be used please

Reply:
Turns out I don't really need to use it after all, but this is good to know. Thanks!